tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/remoting/ReconnectTest.java

```diff
@@ -361,7 +361,7 @@ public boolean intercept(Packet packet, RemotingConnection connection) throws Ac
 
       final long retryInterval = 50;
       final double retryMultiplier = 1d;
-      final int reconnectAttempts = 10;
+      final int reconnectAttempts = 1;
```

Comment:
this is the only test you changed... would this be enough to replicate the issue?


shouldn't we parameterize the test to add this condition, since this seems to be changing other tests...


I need some context on why you changed this.. if this is your test or what is this?


thanks!

Reply:
this test never used the reconnectAttempts because on a failed connection while processing a failover we immediatelly set the FailoverEventType.FAILOVER_FAILED. With my changes we now retry to reconnect for as often as reconnectAttempts is set, therefore on second try we would succeed to reconnect and never send the FailoverEventType.FAILOVER_FAILED.

This test was not done by me but shows that we do not try to reconnect for the specified amount of times.

Adding a test for the issue I wanted to solve seems rather tricky because the issue is only present when the client can connect to the broker and while trying to handle the failover on the session looses the new connection.